django/contrib/contenttypes/fields.py

```diff
@@ -365,7 +365,17 @@ def _get_path_info_with_parent(self):
         opts = self.remote_field.model._meta.concrete_model._meta
         parent_opts = opts.get_field(self.object_id_field_name).model._meta
         target = parent_opts.pk
-        path.append(PathInfo(self.model._meta, parent_opts, (target,), self.remote_field, True, False))
+        path.append(
+            PathInfo(
+                from_opts=self.model._meta,
+                to_opts=parent_opts,
+                target_fields=(target,),
+                join_field=self.remote_field,
+                m2m=True,
+                direct=False,
+                filtered_relation=filtered_relation,
```

Comment:
I think tests for `filtered_relation` and `GenericRelation` are missing. I don't see any failures if this is changed to `filtered_relation=None`.

Reply:
On it

Reply:
I just pushed a commit covering `FilteredRelation` on a generic_foreign_key with a passing test.
But it doesn't cover this specific area of code.

To cover this specific area code we need to have a model that use a GFK defined on a parent model.
example:
```python
class Person(models.Model):
    content_type = models.ForeignKey(
        ContentType,
        on_delete=models.CASCADE,
        null=True,
    )
    object_id = models.PositiveIntegerField(null=True)
    content_object = GenericForeignKey()

class Author(Person):
    pass
```

 Then the FilteredRelation doesn't work. (Generated SQL is using an alias that is not map to a table)
```python
>>> Book.objects.annotate(
...        co_authored_book=FilteredRelation(
...            'co_author',
...            condition=Q(co_author__isnull=False)
...       )
...   ).filter(co_authored_book__isnull=False)
Traceback
...
django.db.utils.OperationalError: no such column: T4.person_ptr_id
```
I will document this limitation. But I'm afraid raising a proper error to warn the user that this edge case is not supported will add more overhead that it is necessary.
Maybe fixing the bug will be easier.
I'd like to think it shouldn't block `FilteredRelation` to land in master and get more exposure from more users.
I don't think `FilteredRelation` is completely bug free, but I believe it is time to have it in master and then fix all corner cases individually as they come up, with smaller chunks during alpha and beta period.
We can have it ready by the final release of 2.0.